src/utils/firestoreUtils.ts

```diff
@@ -87,6 +87,27 @@ export function deleteBuilding(
   });
 }
 
+export function getNameFirestore(uid: string): string {
+  const docRef = firebase.firestore().collection("users").doc(uid);
+  let name = "";
+
+  docRef
+    .get()
+    .then((doc) => {
+      if (doc.exists) {
+        name = doc.data()?.name;
+      } else {
+        // doc.data() will be undefined in this case
+        console.log("No such document!");
+      }
+    })
+    .catch((error) => {
+      console.log("Error getting document:", error);
+    });
+
+  return name;
+}
+
```

Comment:
You should have `getNameFirestore` return this promise chain. Otherwise, I think `name` is always going to get returned as `""`.

```suggestion
export function getNameFirestore(uid: string): Promise<string> {
  const docRef = firebase.firestore().collection("users").doc(uid);
  return docRef
    .get()
    .then((doc) => {
      if (doc.exists) {
        return doc.data()?.name;

      console.log(`No user in "users" with uid ${uid}`);
      return "";
    })
    .catch((error) => {
      console.log(`Error getting data for user ${uid}:`, error);
    });
}

```